### PR TITLE
Improve input/output tensor shape mismatch messages

### LINKED ISSUE #998: SavedModel load sees different input tensor shape than exists in the model
An attempt to load a SavedModel-formatted ResNet50 model - which was converted from a Keras HDF5 file -  fails because TRTIS v1.8.0 running from the NGC-provided Docker image, is seeing a different shape for the input tensor than is specified in the config.pbtxt, which is based on what's shown by the saved_model_cli probe of the model, as well as the original Keras model information.

The saved_model_cli command shows the model's input tensor shape as [-1, -1, -1, 3] as you can see here:

```
MetaGraphDef with tag-set: 'serve' contains the following SignatureDefs:

signature_def['serving_default']:
  The given SavedModel SignatureDef contains the following input(s):
    inputs['resnet50_input:0'] tensor_info:
        dtype: DT_FLOAT
        shape: (-1, -1, -1, 3)
        name: resnet50_input:0
  The given SavedModel SignatureDef contains the following output(s):
    outputs['dense_2/Softmax:0'] tensor_info:
        dtype: DT_FLOAT
        shape: (-1, 5)
        name: dense_2/Softmax:0
  Method name is: tensorflow/serving/predict
```

I set up the config.pbtxt accordingly:

```
input [
  {
    name: "resnet50_input:0"
    data_type: TYPE_FP32
    dims: [ -1, -1, -1, 3 ]
  }
]
```

However, when TRTIS attempts to load this model, it gets the wrong shape for the input tensor, namely [ -1, 3 ], and sees this as a conflict with what's specified in the config.pbtxt:

> model_repository_manager.cc:810] failed to load 'TR46_ResNet50_FFS_300ep0.6train_5classes_noshuffle' version 1: Invalid argument: unable to load model 'TR46_ResNet50_FFS_300ep0.6train_5classes_noshuffle', **tensor 'resnet50_input:0' shape expected by framework [-1,3] doesn't match model configuration shape [-1,-1,-1,3]**

Curiously, when I set the config.pbtxt to use the input dimensions that the above error shows, I get the same kind of error:

> model_repository_manager.cc:810] failed to load 'TR46_ResNet50_FFS_300ep0.6train_5classes_noshuffle' version 1: Invalid argument: unable to load model 'TR46_ResNet50_FFS_300ep0.6train_5classes_noshuffle', **tensor 'resnet50_input:0' shape expected by framework [-1,3] doesn't match model configuration shape [-1,3]**

The logical inconsistency in this error message is what prompted me to open an issue.

When I move the config.pbtxt file out of the way, and run with strict-model-config turned off, it is able to generate a config and load the model successfully, and shows the proper dimensions in the api/status call:

```
      max_batch_size: 1
      input {
        name: "resnet50_input:0"
        data_type: TYPE_FP32
        dims: -1
        dims: -1
        dims: 3
      }
```

However, as described by this passage in the documentation:

> [**Generated Model Configuration**](https://docs.nvidia.com/deeplearning/sdk/tensorrt-inference-server-guide/docs/model_configuration.html)
> Some TensorFlow SavedModel models do not require a model configuration file. The models **must specify all inputs and outputs as fixed-size tensors** (with an optional initial batch dimension) for the model configuration to be generated automatically. 

Predictably, an attempt to use this model comes back with the error "variable-size dimension in model input not supported" and so the model can't be used.

I'd appreciate any suggestions, and would be happy to provide whatever information I'm able.

### PR DESCRIPTION
**Pull request by deadeyegoodwin**
Fix #998

### CODE REVIEW

File: src/backends/onnx/onnx_utils.cc [hunk 1]
```diff
@@ -290,48 +290,63 @@ CompareDimsSupported(
...
-      continue;
+    if (!succ) {
+      return Status(
+          RequestStatusCode::INVALID_ARG,
+          "model '" + model_name + "', tensor '" + tensor_name +
+              "': the model expects " + std::to_string(model_shape.size()) +
+              " dimensions (shape " + DimsListToString(model_shape) +
+              ") but the model configuration specifies " +
+              std::to_string(full_dims.size()) +
+              " dimensions (an initial batch dimension for max_batch_size > 0 "
```
Comment (CoderHam) [inline src/backends/onnx/onnx_utils.cc]: since instead of for

File: src/backends/onnx/onnx_utils.cc [hunk 2]
```diff
@@ -290,48 +290,63 @@ CompareDimsSupported(
...
+              "plus the explicit tensor shape for a full shape " +
```
Comment (CoderHam) [inline src/backends/onnx/onnx_utils.cc]: _followed by the explicit tensor shape, making the complete shape_

File: src/backends/tensorflow/tf_utils.cc [hunk 3]
```diff
@@ -29,92 +29,70 @@
...
           RequestStatusCode::INVALID_ARG,
-          "the model expects shape " + ShapeToString(model_shape) +
-              " but the model configuration specified shape " +
-              DimsListToString(dims));
+          "model '" + model_name + "', tensor '" + tensor_name +
+              "': the model expects " + std::to_string(model_shape->rank_) +
+              " dimensions (shape " + ShapeToString(model_shape) +
+              ") but the model configuration specifies " +
+              std::to_string(full_dims.size()) +
+              " dimensions (an initial batch dimension for max_batch_size > 0 "
```
Comment (CoderHam) [inline src/backends/tensorflow/tf_utils.cc]: since instead of for

File: src/backends/tensorflow/tf_utils.cc [hunk 4]
```diff
@@ -29,92 +29,70 @@
...
+              "plus the explicit tensor shape for a full shape " +
```
Comment (CoderHam) [inline src/backends/tensorflow/tf_utils.cc]: followed by the explicit tensor shape, making the complete shape

File: src/backends/tensorrt/plan_utils.cc [hunk 5]
```diff
@@ -136,42 +136,63 @@ CompareDimsSupported(
...
+    if (!succ) {
+      return Status(
+          RequestStatusCode::INVALID_ARG,
+          "model '" + model_name + "', tensor '" + binding_name +
+              "': the model expects " + std::to_string(model_dims.nbDims) +
+              " dimensions (shape " + DimsDebugString(model_dims) +
+              ") but the model configuration specifies " +
+              std::to_string(full_dims.size()) +
+              " dimensions (an initial batch dimension for max_batch_size > 0 "
+              "plus the explicit tensor shape for a full shape " +
```
Comment (CoderHam) [inline src/backends/tensorrt/plan_utils.cc]: followed by the explicit tensor shape, making the complete shape

File: src/backends/tensorrt/plan_utils.cc [hunk 6]
```diff
@@ -136,42 +136,63 @@ CompareDimsSupported(
...
-      continue;
+    if (!succ) {
+      return Status(
+          RequestStatusCode::INVALID_ARG,
+          "model '" + model_name + "', tensor '" + binding_name +
+              "': the model expects " + std::to_string(model_dims.nbDims) +
+              " dimensions (shape " + DimsDebugString(model_dims) +
+              ") but the model configuration specifies " +
+              std::to_string(full_dims.size()) +
+              " dimensions (an initial batch dimension for max_batch_size > 0 "
```
Comment (CoderHam) [inline src/backends/tensorrt/plan_utils.cc]: Since instead of for

Comment (CoderHam) [reply src/backends/tensorrt/plan_utils.cc]: Change to dimension because max_batch_size to make consistent with other changes

Comment (deadeyegoodwin) [reply src/backends/tensorrt/plan_utils.cc]: Fixed

**Pull request merged**